contents/tutorials/nuxt-feature-flags.md

```diff
@@ -0,0 +1,247 @@
+    }
+  }
+})
+```
+
+Create a new [plugin](https://nuxt.com/docs/guide/directory-structure/plugins) by creating a new folder called `plugins` in your base directory and then a new file `posthog.client.js`:
+
+```bash
+mkdir plugins
+cd plugins 
+touch posthog.client.js
+```
+
+Add the following code to your `posthog.client.js` file:
+
+```js file=plugins/posthog.client.js
+import { defineNuxtPlugin } from '#app'
+import posthog from 'posthog-js'
+
+export default defineNuxtPlugin(nuxtApp => {
+  const runtimeConfig = useRuntimeConfig();
+  const posthogClient = posthog.init(runtimeConfig.public.posthogPublicKey, {
+    api_host: runtimeConfig.public.posthogHost,
+  })
+  
+  return {
+    provide: {
+      posthog: () => posthogClient
+    }
+  }
+})
+```
+
+Once you’ve done this, reload your app. You should begin seeing events in the [PostHog events explorer](https://us.posthog.com/events).
+
+<ProductScreenshot
+  imageLight={EventsLight} 
+  imageDark={EventsDark} 
+  alt="Events in PostHog" 
+  classes="rounded"
+/>
+
+## Creating a feature flag
+
+With PostHog set up, your app is ready for feature flags. To create one, go to the [feature flags tab](https://us.posthog.com/feature_flags) in PostHog and click "New feature flag." Enter a flag key (like `my-cool-flag`), set the release condition to roll out to 100% of users, and press "Save."
+
+<ProductScreenshot
+  imageLight={CreateFlagLight} 
+  imageDark={CreateFlagDark} 
+  alt="Feature flag created in PostHog" 
+  classes="rounded"
+/>
+
+You can customize your [release conditions](/docs/feature-flags/creating-feature-flags#release-conditions) with rollout percentages, and [user](/docs/product-analytics/user-properties) or [group properties](/docs/product-analytics/group-analytics) to fit your needs.
+
+## Implementing flag code on the client side
+
+Once created, we can add our feature flag to our app. We do this using the [`posthog.onFeatureFlags`](/docs/libraries/js#ensuring-flags-are-loaded-before-usage) callback.
+
+In our example, we fetch our feature flag and update the text on the page based on its value:
+
+```vue file=app.vue
+<template>
+  <main>
+    <h1>{{ title }}</h1>
+  </main>
+</template>
+
+<script setup>
+
+const title = ref('No variant');
+onMounted(() => {
+  const { $posthog } = useNuxtApp()
+  if ($posthog) {
+    const posthog = $posthog()
+    if (posthog.isFeatureEnabled('my-cool-flag')) {
+      title.value = 'Our flag is enabled!';
+    } else {
+      title.value = 'Our flag is disabled!';
+    }
+  }
+});
+
+</script>
+```
+
+When you run your app now, you should see the updated text.
+
+![New app after adding the flag](../images/tutorials/nuxt-feature-flags/flag-enabled-in-app.png)
+
+## Implementing flag code on the server side
+
+You may notice the text flickers when you load your app. This is because PostHog is making a request on the client side to fetch the flag value. To prevent this, there are two alternatives:
```

Comment:
```suggestion
You may notice the text flickers when you load your app. This is because PostHog is making a request on the client side to fetch the flag value. There are two ways to prevent this:
```